src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1275,9 +1313,97 @@ bool entitiesQuery
         cerV->push_back(cer);
     }
 
+    /* All the not-patterned entities in the request not in the response are added (without attributes), as they are
+     * used before prunning in the CPr calculation logic */
+    for (unsigned int ix = 0; ix < enV.size(); ++ix)
+    {
+      if (enV.get(ix)->isPattern != "true")
+      {
+        bool needToAdd = true;
+        for (unsigned int jx = 0; jx < cerV->size(); ++jx)
+        {
+          if (cerV->get(jx)->contextElement.entityId.id == enV.get(ix)->id && cerV->get(jx)->contextElement.entityId.type == enV.get(ix)->type)
+          {
+            needToAdd = false;
+            break;  /* jx */
+          }
+        }
+        if (needToAdd)
+        {
+          ContextElementResponse* cerP = new ContextElementResponse();
+          cerP->contextElement.entityId.id = enV.get(ix)->id;
+          cerP->contextElement.entityId.type = enV.get(ix)->type;
+          cerP->contextElement.entityId.isPattern = "false";
+
+          /* This entity has to be prunned if after CPr searching no attribute is "added" to it. The notPrune attribute distinguish this kind of
+           * entities from the "real ones" that are without attributes in the Orion local database (a rare case, but may happen) */
+          cerP->notPrune = false;
+
+          for (unsigned int jx = 0; jx < attrL.size(); ++jx)
+          {
+            ContextAttribute* caP = new ContextAttribute(attrL.get(jx), "", "");
+            caP->found = false;
+            cerP->contextElement.contextAttributeVector.push_back(caP);
+          }
+
+          cerP->statusCode.fill(SccOk);
+
+          cerV->push_back(cerP);
+        }
+      }
+    }
+
     return true;
 }
 
+/* ****************************************************************************
+*
+* pruneNotFoundContextElements -
+*
+* Remove attributes in the vector with 'found' value is 'false'
+*
+*/
+void pruneNotFoundContextElements(ContextElementResponseVector& oldCerV, ContextElementResponseVector* newCerVP)
+{
+  for (unsigned int ix = 0; ix < oldCerV.size(); ++ix)
+  {
+    ContextElementResponse* cerP = oldCerV.get(ix);
+    ContextElementResponse* newCerP = new ContextElementResponse();
+
+    /* Note we cannot use the ContextElement::fill() method, given that it also copies the ContextAttributeVector. The side-effect
+     * of this is that attributeDomainName and domainMetadataVector and not being copied, but it should not be a problem, given that
+     * domain attributes are not implemented */
+    newCerP->contextElement.entityId.fill(&cerP->contextElement.entityId);
+    newCerP->contextElement.providingApplicationList = cerP->contextElement.providingApplicationList;  // FIXME P10: not sure if this is the right way of doing, maybe we need a fill() method for this
```

Comment:
This would involve to become providingApplicationList into a class. Do you think is a good idea? A bit overwhelming for something that is basically a vector of strings and nothing more...